### PR TITLE
Add solution analysis display (#97)

### LINKED ISSUE #97: Add minimal needed solution analysis to SlicerOpenLIFU
Put the solution analysis aspects into SlicerOpenLIFU enough to just update the pulse in the solution object (so that the hardware can be told what to do)

This should be part of what automatically executes when we click "plan"

Depends on #96 and https://github.com/OpenwaterHealth/OpenLIFU-python/issues/104

### PR DESCRIPTION
**Pull request by ebrahimebrahim**
Close #97 

## changes

- Solution analysis is displayed in two tables in the sonication planning module
- If there's no solution, then there's also no solution analysis, and the widget will just say "No solution" instead of showing blank tables.
- Once a solution is computed, the analysis is set. It's in the planning module parameter node.
- The first table has a column for each focus point, while the second table shows the metrics that are "global" rather than per-focus-point in just one column.
- If the solution is removed (e.g. by changing or unloading session) then the analysis is unset.
- Given the current application logic, it should not be possible for there to be a solution in the scene and have the solution analysis variable get cleared out (set to None). However I am worried that it might happen in some situations what with all the power that the manual workflow provides, so if that does happen then the analysis gets recomputed. Just to be safe. To test this behavior you can trigger the hopefully not possible state by doing the following in the python console after computing a solution: `slicer.modules.OpenLIFUSonicationPlannerWidget.logic.getParameterNode().solution_analysis = None`.

## remaining todo 

- [x] await the merging of https://github.com/OpenwaterHealth/OpenLIFU-python/pull/155
- [x] update the python-requirements.txt
- [x] await merging of https://github.com/OpenwaterHealth/OpenLIFU-python/pull/157
- [x] update the python-requirements.txt
- [x] I always get something very close to 1 for `mainlobe_pnp_MPa`. Maybe because of solution scaling, but is that how it's supposed to work? Check with Peter, and if it's not how it's supposed to work open a bug on OpenLIFU-python
  - this is likely related to the regression https://github.com/OpenwaterHealth/OpenLIFU-python/issues/141
- [x] don't just show blank tables but rather display an error label in the analysis widget when it ends up in an error state
- [x] put the two tables into groupboxes  with headings that explain one is per-focus and the other is not.
- [x] turn off `verticalHeaderVisible` for the tables

### CODE REVIEW

**Comment by ebrahimebrahim** [comment]
@sadhana-r There are some TODO's but not much more should need to change, just the openlifu-python version.

**Comment by ebrahimebrahim** [comment]
(the force-push just now was only to change python-requirements.txt)

File: OpenLIFUSonicationPlanner/OpenLIFUSonicationPlanner.py [hunk 1]
```diff
@@ -123,9 +133,11 @@ def setup(self) -> None:
         self.updateRenderPNPCheckBox()
         self.updateVirtualFitApprovalStatus()
         self.updateTrackingApprovalStatus()
+        self.updateSolutionAnalysis()
 
-        # Add an observer on the Data module's parameter node
+        # Add observers on the Data module's parameter node and this module's own parameter node
         self.addObserver(get_openlifu_data_parameter_node().parameterNode, vtk.vtkCommand.ModifiedEvent, self.onDataParameterNodeModified)
+        self.addObserver(self.logic.getParameterNode().parameterNode, vtk.vtkCommand.ModifiedEvent, self.onParameterNodeModified)
```
[the remark this answers is not in the document]
Comment (ebrahimebrahim) [reply OpenLIFUSonicationPlanner/OpenLIFUSonicationPlanner.py:140]: Oh... huh. That would be much better. Would be nice to reorganize and adopt these conventions. I'll open a separate issue!

File: OpenLIFUSonicationPlanner/OpenLIFUSonicationPlanner.py [hunk 2]
```diff
@@ -333,6 +348,109 @@ def onApproveClicked(self):
...
+
+        # Max length of list type fields in the dataclass
+        max_len = max(
+            len(getattr(analysis_openlifu, f.name))
+            for f in fields(analysis_openlifu)
+            if get_origin(f.type) is list
+        )
+
+        self.focusAnalysisTableModel.setHorizontalHeaderLabels(['Metric'] + [f"Focus {i+1}" for  i in range(max_len)])
+        self.globalAnalysisTableModel.setHorizontalHeaderLabels(['Metric', 'Value'])
```
[the remark this answers is not in the document]
Comment (ebrahimebrahim) [reply OpenLIFUSonicationPlanner/OpenLIFUSonicationPlanner.py:415]: ah, good ideas, I think I prefer the second suggestion

**Review by sadhana-r — APPROVED** [review]

**Pull request merged**